### PR TITLE
feat: extended registry metadata response

### PR DESCRIPTION
**Pull request by spalt08**
- More useful fields for the package metadata entity
- Updated process for the package sync
- Skip flaky tests with the NPM registry

### CODE REVIEW

File: packages/worker/src/npmRegistry/api.ts [hunk 1]
```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
...
+  }
+
+  // (git+)https://github.com/facebook/react(.git)
+  if ((url = repository.match(/^(|git\+)(https?:.*?)(|.git)$/))) {
+    return url[2];
+  }
+
+  // git://github.com/auth0/lock(.git)
+  if ((url = repository.match(/^git:\/\/(.*?)(|.git)$/))) {
+    return `https://${url[2]}`;
```
Comment (ctizen) [inline packages/worker/src/npmRegistry/api.ts:120]: Did you mean `url[1]` here?

File: packages/worker/src/npmRegistry/api.ts [hunk 2]
```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
...
+
+function getRepositoryUrl(repository?: string) {
+  let url;
+
+  if (!repository) {
+    return;
+  }
+
+  // (git+)https://github.com/facebook/react(.git)
+  if ((url = repository.match(/^(|git\+)(https?:.*?)(|.git)$/))) {
```
Comment (ctizen) [inline packages/worker/src/npmRegistry/api.ts:114]: Some considerations for the regexps here and below:
- You could use non-capturing groups (`(?:...)`) which don't get into the output array. 
- You didn't escape the dot metasymbol in the last group.
- Constructs like `(|smth)` are confusing for me, I'd prefer it to be `(smth)?`, and considering first advice about non-capturing groups, `(?:smth)?`.
- The proper equivalent of the expression with fixes above included would be something like `/^(?:git\+)?(https?:.*?)(?:\.git)?$/`

Comment (ctizen) [reply packages/worker/src/npmRegistry/api.ts:114]: Disclaimer: all written above was not tested at all :) https://regex101.com for the rescue.

File: packages/worker/src/npmRegistry/api.ts [hunk 3]
```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
...
+  try {
+    url = new URL(maybeUrl);
+  } catch (_) {
+    return false;
+  }
+
+  return url.protocol === 'http:' || url.protocol === 'https:';
+}
+
+function getRepositoryUrl(repository?: string) {
```
Comment (ctizen) [inline packages/worker/src/npmRegistry/api.ts:106]: Probably there should be some tests covering this function.

**Review by ctizen — CHANGES_REQUESTED** [review]

**Review by ctizen — APPROVED** [review]

File: packages/worker/src/npmRegistry/api.ts [hunk 4]
```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
...
+
+export function getRepositoryUrl(repository?: string) {
+  let url;
+
+  if (!repository) {
+    return;
+  }
+
+  // (git+)https://github.com/facebook/react(.git)
+  if ((url = repository.match(/^(?:git\+)?(https?:.*?)(?:\.git)?$/))) {
```
Comment (zardak) [inline packages/worker/src/npmRegistry/api.ts:114]: nit: When working with capture groups, destructuring can bring clarity to what's being extracted, like:
```
const urlMatch = repository.match(/^(?:git\+)?(https?:.*?)(?:\.git)?$/);
if (urlMatch) {
  const [_, reporsitoryUrl] = urlMatch;
  ...
}
```

**Review by zardak — APPROVED** [review]

File: packages/worker/src/npmRegistry/api.ts (hunk 4 above)
Comment (spalt08) [reply packages/worker/src/npmRegistry/api.ts]: good point, fixed

**Pull request merged**